### PR TITLE
Disable microphone and camera buttons when there are no microphones or cameras

### PR DESCRIPTION
**Pull request by JamesBurnside**
# What
* Update camera and mic button selectors to make the buttons disabled when there are no cameras or mic available

# Why
They shouldn't be pressable in those cases - causes odd cannot start microphone errors - ideally we should also provide a tooltip showing why they are disabled but we show an error UFD there are no microphones so that should suffice and avoid a lot of extra logic

# How Tested
See UI tests added.

### CODE REVIEW

File: packages/react-composites/tests/browser/call/app/mocks/MockCallAdapter.ts [hunk 1]
```diff
@@ -48,6 +49,8 @@ export class MockCallAdapter implements CallAdapter {
       initialState.latestErrors = testState.latestErrors;
     }
 
+    initialState.devices = merge(initialState.devices, testState.devices);
```
Comment (mgamis-msft) [inline packages/react-composites/tests/browser/call/app/mocks/MockCallAdapter.ts:52]: Cool. I did not know fluent/react has a deep merge method.

**Review by mgamis-msft — APPROVED** [review]

**Review by dmceachernmsft — APPROVED** [review]

File: packages/calling-component-bindings/src/callControlSelectors.ts [hunk 2]
```diff
@@ -43,7 +43,7 @@ export const microphoneButtonSelector: MicrophoneButtonSelector = reselect.creat
   (callExists, isMuted, deviceManager) => {
     const permission = deviceManager.deviceAccess ? deviceManager.deviceAccess.audio : true;
     return {
-      disabled: !callExists || !permission,
+      disabled: !callExists || !permission || !deviceManager.microphones.length,
```
Comment (anjulgarg) [inline packages/calling-component-bindings/src/callControlSelectors.ts:46]: I am assuming the drop down still works allowing users to change their speakers?

**Review by anjulgarg — APPROVED** [review]

File: packages/calling-component-bindings/src/callControlSelectors.ts (hunk 2 above)
Comment (anjulgarg) [inline packages/calling-component-bindings/src/callControlSelectors.ts:46]: does this still allow the speakers to be changes using the microphone dropdown? (callwithchat)

Comment (JamesBurnside) [reply packages/calling-component-bindings/src/callControlSelectors.ts]: Great catch! Updated: 
image

**Pull request merged**